tycho-bundles/org.eclipse.tycho.p2.resolver.impl/src/main/java/org/eclipse/tycho/p2/target/TargetPlatformFactoryImpl.java

```diff
@@ -470,52 +454,82 @@ static int countElements(Iterator<?> iterator) {
             throw new IllegalArgumentException(
                     "Base target platform must be an instance of PreliminaryTargetPlatformImpl; was: "
                             + baseTargetPlatform);
         }
+        if (!(pomDependencyCollector instanceof PomDependencyCollectorImpl)) {
+            throw new IllegalArgumentException(
+                    "PomDependencyCollector must be an instance of PomDependencyCollectorImpl; was: "
+                            + pomDependencyCollector);
+        }
         return createTargetPlatformWithUpdatedReactorUnits((PreliminaryTargetPlatformImpl) baseTargetPlatform,
-                reactorUnits, reactorArtifacts);
+                reactorUnits, reactorArtifacts, (PomDependencyCollectorImpl) pomDependencyCollector);
     }
 
     P2TargetPlatform createTargetPlatformWithUpdatedReactorUnits(PreliminaryTargetPlatformImpl preliminaryTP,
             Map<IInstallableUnit, ReactorProjectIdentities> reactorUnitsMap,
-            List<IRawArtifactFileProvider> reactorArtifacts) {
-
+            List<IRawArtifactFileProvider> reactorArtifacts, PomDependencyCollectorImpl pomDependencyCollector) {
         LinkedHashSet<IInstallableUnit> allUnits = preliminaryTP.getExternalUnits();
-
+        TargetPlatformFilterEvaluator configuredFilters = preliminaryTP.getFilter();
         // external units are already filtered, only reactor units need to be filtered again
         if (reactorUnitsMap != null) {
-            Collection<IInstallableUnit> reactorUnits = new LinkedList<>(reactorUnitsMap.keySet());
-
-            TargetPlatformFilterEvaluator configuredFilters = preliminaryTP.getFilter();
-            if (configuredFilters != null) {
-                configuredFilters.filterUnits(reactorUnits);
-            }
-            allUnits.addAll(reactorUnits);
+            allUnits.addAll(filterUnits(configuredFilters, reactorUnitsMap.keySet()));
         }
 
+        IRawArtifactFileProvider pomDependencyArtifactRepo = pomDependencyCollector.getArtifactRepoOfPublishedBundles();
         IRawArtifactFileProvider jointArtifacts = createJointArtifactProvider(reactorArtifacts,
-                preliminaryTP.getExternalArtifacts());
+                preliminaryTP.getExternalArtifacts(), pomDependencyArtifactRepo);
+        RepositoryBlackboardKey blackboardKey = RepositoryBlackboardKey
+                .forResolutionContextArtifacts(pomDependencyCollector.getProjectLocation());
+        ArtifactRepositoryBlackboard.putRepository(blackboardKey, new ProviderOnlyArtifactRepository(jointArtifacts,
+                Activator.getProvisioningAgent(), blackboardKey.toURI()));
+        logger.debug("Registered artifact repository " + blackboardKey);
+        //FIXME map POM deps to reactor ReactorProjectIdentities??
+        allUnits.addAll(filterUnits(configuredFilters, pomDependencyCollector.gatherMavenInstallableUnits()));
+
+        logger.debug("Registered artifact repository " + blackboardKey);
```

Comment:
If this `createTargetPlatformWithUpdatedReactorUnits` called for all modules?
If yes, I suggest we remove the existing similar part that does create the RepositoryBlackboard in `createTargetPlatform`, as it's now meant to not be a complete result, thus not suitable for director/blackboard.

Reply:
This is already moved from createTargetPlatform > createTargetPlatformWithUpdatedReactorUnits and called in the TargetPlatform stage of the build.

Reply:
OK, perfect. I missed it.
Note that this debug line is shown twice. You probably want to get rid of one of the 2 occurences